pyhive/hive.py

```diff
@@ -252,6 +258,41 @@ def _fetch_more(self):
         for row in response.results.rows:
             self._data.append([_unwrap_col_val(val) for val in row.colVals])
 
+    def poll(self, fetch_logs=False):
+        req = ttypes.TGetOperationStatusReq(
+            operationHandle=self._operationHandle
+        )
+        response = self._connection.client.GetOperationStatus(req)
+        _check_status(response)
+        
+        
+        
+        if fetch_logs:
```

Comment:
I think this is cleaner as a separate method, e.g. `def fetch_logs(): List[String]`